login_server/loginserver.py

```diff
@@ -351,9 +352,42 @@ def handle_http_request_message(self, msg):
                 'online_players_list': [p.display_name for p in self.players.values()],
                 'online_servers_list': online_game_servers_list
             }, sort_keys=True, indent=4))
+        elif msg.env['PATH_INFO'] == '/player':
+            if "REMOTE_ADDR" in msg.env:
+                self.logger.info('Served player stats request via HTTP to peer "' + msg.env["REMOTE_ADDR"] + '"')
+            else:
+                self.logger.info('Served player stats request via HTTP to Unknown peer')
+            
+            if "QUERY_STRING" in msg.env:
+                player_data = self.get_player_settings_data(msg.env['QUERY_STRING'])
+                if player_data != False:
+                    filtered_player_data = {
+                        "player_found": True,
+                        "clan_tag": player_data["clan_tag"],
+                        "player_name": msg.env['QUERY_STRING'],
+                        "rank_xp": player_data["progression"]["rank_xp"]
+                    }
+                    msg.peer.send_response(json.dumps(filtered_player_data, sort_keys=True, indent=4))
+                else:
+                    msg.peer.send_response(json.dumps({
+                        'player_found': False
+                    }, sort_keys=True, indent=4))
+            else:
+                msg.peer.send_response(json.dumps({
+                    'player_found': False
+                }, sort_keys=True, indent=4))
         else:
             msg.peer.send_response(None)
 
+    def get_player_settings_data(self, player_name):
+        self.logger.info("Retrieved settings data for player: " + player_name)
+        try:
+            with open(glob.glob('data/players/' + player_name + '_settings.json')[0], "r") as f:
```

Comment:
Why would you use glob here when you already know the filename? Do you want people to pass wildcards?

Using input from a user to determine a file to read is very unsafe. What if I were to specify "../gamesettings/ootb/serverconfig.lua\0" in order to read the admin password. I know it requires more to really do anything harmful, but this is the first step.

The input data should not be trusted. So we could check if it's a valid username by checking if it is in self.accounts.